### DISCUSSION TITLE
Test Driven Development

### DISCUSSION THREAD

**Discussion by Super-Nim** [Encode-Solidity-Bootcamp-Group-10/Week2#1] [General]
Who has the most experience with TDD? 🤔 

I understand we didn't _need_ to write our own contracts for week 2, and for week 3. However, I'm keen to learn this skill and want to see what the general consensus is on this method?

For those unfamiliar, we write down the essential functionality and create unit tests for them. The unit test would follow the [Arrange Act Assert](https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/) (AAA) format.

[A Solidity article on TDD](https://michalzalecki.com/ethereum-test-driven-introduction-to-solidity/)

**Comment by IlyaSinyugin** [comment]
I do not have a lot of experience with TDD. But I am keen to apply this skill in our upcoming projects since testing smart contracts is very crucial.

Also, if someone wants to refresh their memory on TDD and its benefits, have a look at this [article](https://www.browserstack.com/guide/what-is-test-driven-development) which is linked as one of the references for the 3rd lecture.

**Reply by vishalavalani** [reply]
Same with me, in most of my projects we don't have TDD.
